cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -215,14 +215,22 @@ def gather_sdn_data(sdn_plugin):
     set_state('kube-sdn.configured')
 
 
-@when('kube_master_components.installed')
+@when('config.changed.dashboard', 'kubernetes.dashboard.available')
+def reset_states():
+    remove_state('kubernetes.dashboard.available')
+    launch_kubernetes_dashboard()
+
+
+@when('kube-dns.available')
 @when_not('kubernetes.dashboard.available')
 def launch_kubernetes_dashboard():
     ''' Launch the Kubernetes dashboard. If not enabled, attempt deletion '''
     if hookenv.config('dashboard'):
-        # TODO - make this self contained
-        dashboard_manifest = 'https://rawgit.com/kubernetes/dashboard/master/src/deploy/kubernetes-dashboard.yaml'  # noqa
-        cmd = ['kubectl', 'create', '-f', dashboard_manifest]
+        context = {}
+        context['arch'] = arch()
+        manifest = '/etc/kubernetes/addons/dashboard.yaml'
+        render('kubernetes-dashboard.yaml', manifest, context)
+        cmd = ['kubectl', 'create', '-f', manifest]
         call(cmd)
         set_state('kubernetes.dashboard.available')
     else:
```

Comment:
Can you change this to delete the dashboard by the same file? That way you don't need to "know" the namespace or the name.